sky/templates/gcp-ray.yml.j2

```diff
@@ -96,10 +102,10 @@ available_node_types:
         - onHostMaintenance: TERMINATE  # Required for GPU-attached VMs.
   {%- endif %}
 {%- endif %}
-{% if num_nodes > 1 %}
+{% if num_nodes - num_reserved_nodes > 1 %}
   ray_worker_default:
-    min_workers: {{num_nodes - 1}}
-    max_workers: {{num_nodes - 1}}
+    min_workers: {{num_nodes - num_reserved_nodes - 1}}
+    max_workers: {{num_nodes - num_reserved_nodes - 1}}
```

Comment:
Sorry, my bad. This is a bit trickier. This probably need to be:
```
{% if num_nodes - 1 - num_reserved_workers > 0 %}
  ray_worker_default:
    min_workers: {{num_nodes - 1 - num_reserved_workers}}
    max_workers: {{num_nodes - 1 - num_reserved_workers}}
```

The `num_reserved_workers` should be set to `num_reserved_workers = max(num_reserved_nodes - 1, 0)` in the python script.

Reply:
done